virtool/analyses/data.py

```diff
@@ -1,24 +1,179 @@
+
+    async def find(self, request: Request) -> List[Union[dict, List[AnalysisMinimal]]]:
+        """
+        List all analysis documents.
+        """
+        db_query = {}
+
+        data = await paginate(
+            self._db.analyses,
+            db_query,
+            request.query,
+            projection=PROJECTION,
+            sort=[("created_at", -1)],
+        )
+
+        per_document_can_read = await gather(
+            *[
+                virtool.samples.db.check_rights(
+                    self._db, document["sample"]["id"], request["client"], write=False
+                )
+                for document in data["documents"]
+            ]
+        )
+
+        documents = [
+            document
+            for document, can_write in zip(data["documents"], per_document_can_read)
+            if can_write
+        ]
+
+        documents = await apply_transforms(
+            documents,
+            [AttachUserTransform(self._db), AttachSubtractionTransform(self._db)],
+        )
+
+        return [data, documents]
+
+    async def get(self, request: Request) -> List[Union[dict, Sample]]:
+        """
+        Get a single analysis by its ID.
+        """
+        analysis_id = request.match_info["analysis_id"]
 
-    async def get_by_id(self, analysis_id: str) -> Optional[Document]:
         document = await self._db.analyses.find_one(analysis_id)
 
         if document is None:
-            return None
+            raise ResourceNotFoundError()
 
-        if "updated_at" not in document:
-            document["updated_at"] = document["created_at"]
+        try:
+            iso = isoformat(document["updated_at"])
+        except KeyError:
+            iso = isoformat(document["created_at"])
+
+        if_modified_since = request.headers.get("If-Modified-Since")
+
+        if if_modified_since and if_modified_since == iso:
+            raise ResourceNotModifiedError()
+
+        document = await attach_analysis_files(self._pg, analysis_id, document)
+
+        sample = await self._db.samples.find_one(
+            {"_id": document["sample"]["id"]}, {"quality": False}
+        )
+
+        if not sample:
+            raise ResourceError()
+
+        read, _ = get_sample_rights(sample, request["client"])
+
+        if not read:
+            raise InsufficientRights()
 
         if document["ready"]:
-            return await virtool.analyses.format.format_analysis(self._app, document)
+            document = await virtool.analyses.format.format_analysis(
+                request.app, document
+            )
+
+        headers = {
+            "Cache-Control": "no-cache",
+            "Last-Modified": isoformat(document["created_at"]),
+        }
+
+        document = await processor(self._db, document)
+
+        if document["workflow"] == "nuvs":
+            document = await apply_transforms(
+                document,
+                [AttachNuVsBLAST(self._pg)],
+            )
+
+        return [document, headers]
```

Comment:
You are going to remove this, but this will implicitly return a `tuple` and is preferred Python style in this use case:
```py
return documents, headers
```
Correct this elsewhere.